File: Python/6py 1/high.py
"""
|--------------------------------------------------------------------------
| Highest Scoring Word
|--------------------------------------------------------------------------
|
"""

def high(x):
   maxx, res = 0, ""
   for word in x.split(" "):
      score = sum(ord(char) - 96 for char in word)
      if score > maxx: maxx, res = score, word
   return res


def high(x):
   return max(x.split(), key=lambda k: sum(ord(c) - 96 for c in k))


def high(x):
   words=x.split(' ')
   list = []
   for i in words:
      scores = [sum([ord(char) - 96 for char in i])]
      list.append(scores)
   return words[list.index(max(list))]


print(high('man i need a taxi up to ubud'), 'taxi')
print(high('what time are we climbing up the volcano'), 'volcano')
print(high('take me to semynak'), 'semynak')
print(high('aa b'), 'aa')
print(high('b aa'), 'b')
print(high('bb d'), 'bb')
print(high('d bb'), 'd')
print(high("aaa b"), "aaa")

File: JavaScript/6js 1/high.js
/*
|--------------------------------------------------------------------------
| Highest Scoring Word
|--------------------------------------------------------------------------
|
*/

function high(x) {
   let max = 0;
   let res = "";
   for (word of x.split(" ")) {
      let score = 0;
      for (let i = 0; i < word.length; i++) score += word.charCodeAt(i) - 96;
      if (score > max) {
         max = score;
         res = word;
      }
   }
   return res;
}


function high(x) {
   const score = (s) => s.split("").reduce((m, i) => m + i.charCodeAt(0) - 96, 0);
   return x.split(" ").reduce((s, n) => (score(s) < score(n) ? n : s));
}


function high(x) {
   const alpha = "abcdefghijklmnopqrstuvwxyz";
   const words = x.split(" ");
   const scores = words
      .map((x) => [...x].map((y) => alpha.indexOf(y) + 1))
      .map((x) => x.reduce((a, b) => a + b, 0));
   return words[scores.indexOf(Math.max(...scores))];
}


console.log(high("man i need a taxi up to ubud"), "taxi");
console.log(high("what time are we climbing up the volcano"), "volcano");
console.log(high("take me to semynak"), "semynak");
console.log(high("aa b"), "aa");
console.log(high("b aa"), "b");
console.log(high("bb d"), "bb");
console.log(high("d bb"), "d");
console.log(high("aaa b"), "aaa");